cmake/Dependencies.cmake

```diff
@@ -1787,43 +1787,68 @@ list(APPEND Caffe2_DEPENDENCY_LIBS fmt::fmt-header-only)
 set(BUILD_SHARED_LIBS ${TEMP_BUILD_SHARED_LIBS} CACHE BOOL "Build shared libs" FORCE)
 
 # ---[ Kineto
+if(USE_KINETO AND INTERN_BUILD_MOBILE)
+  message(STATUS "Not using libkineto in a mobile build.")
+  set(USE_KINETO OFF)
+endif()
+if(USE_KINETO AND (NOT USE_CUDA))
+  message(STATUS "Not using libkineto in a non-CUDA build.")
+  set(USE_KINETO OFF)
+endif()
+if(USE_KINETO AND MSVC)
+  message(STATUS "Not using libkineto in a Windows build.")
+  set(USE_KINETO OFF)
+endif()
 if(USE_KINETO)
-  if(USE_KINETO AND NOT TARGET kineto)
-    set(CAFFE2_THIRD_PARTY_ROOT "${PROJECT_SOURCE_DIR}/third_party" CACHE STRING "")
-    set(KINETO_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/kineto/libkineto" CACHE STRING "")
-    set(KINETO_BUILD_TESTS OFF CACHE BOOL "")
-    set(KINETO_LIBRARY_TYPE "static" CACHE STRING "")
-    set(CUDA_SOURCE_DIR "${CUDA_TOOLKIT_ROOT_DIR}" CACHE STRING "")
-
-    message(STATUS "Configuring Kineto dependency:")
-    message(STATUS "  KINETO_SOURCE_DIR = ${KINETO_SOURCE_DIR}")
-    message(STATUS "  KINETO_BUILD_TESTS = ${KINETO_BUILD_TESTS}")
-    message(STATUS "  KINETO_LIBRARY_TYPE = ${KINETO_LIBRARY_TYPE}")
-    message(STATUS "  CUDA_SOURCE_DIR = ${CUDA_SOURCE_DIR}")
-
-    if(EXISTS ${CUDA_SOURCE_DIR}/extras/CUPTI/include)
-      set(CUPTI_INCLUDE_DIR "${CUDA_SOURCE_DIR}/extras/CUPTI/include")
-    elseif(EXISTS ${CUDA_SOURCE_DIR}/include/cupti.h)
-      set(CUPTI_INCLUDE_DIR "${CUDA_SOURCE_DIR}/include")
-    endif()
-
-    if((NOT DEFINED CUDA_cupti_LIBRARY) OR (${CUDA_cupti_LIBRARY} STREQUAL "CUDA_cupti_LIBRARY-NOTFOUND"))
-      if(EXISTS ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64/libcupti_static.a)
-        set(CUDA_cupti_LIBRARY "${CUDA_SOURCE_DIR}/extras/CUPTI/lib64/libcupti_static.a")
-      elseif(EXISTS ${CUDA_SOURCE_DIR}/lib64/libcupti_static.a)
-        set(CUDA_cupti_LIBRARY "${CUDA_SOURCE_DIR}/lib64/libcupti_static.a")
-      elseif(EXISTS ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64/libcupti.so)
-        set(CUDA_cupti_LIBRARY "${CUDA_SOURCE_DIR}/extras/CUPTI/lib64/libcupti.so")
-      elseif(EXISTS ${CUDA_SOURCE_DIR}/lib64/libcupti.so)
-        set(CUDA_cupti_LIBRARY "${CUDA_SOURCE_DIR}/lib64/libcupti.so")
-      endif()
+  set(CAFFE2_THIRD_PARTY_ROOT "${PROJECT_SOURCE_DIR}/third_party" CACHE STRING "")
+  set(KINETO_SOURCE_DIR "${CAFFE2_THIRD_PARTY_ROOT}/kineto/libkineto" CACHE STRING "")
+  set(KINETO_BUILD_TESTS OFF CACHE BOOL "")
+  set(KINETO_LIBRARY_TYPE "static" CACHE STRING "")
+  set(CUDA_SOURCE_DIR "${CUDA_TOOLKIT_ROOT_DIR}" CACHE STRING "")
+
+  message(STATUS "Configuring Kineto dependency:")
+  message(STATUS "  KINETO_SOURCE_DIR = ${KINETO_SOURCE_DIR}")
+  message(STATUS "  KINETO_BUILD_TESTS = ${KINETO_BUILD_TESTS}")
+  message(STATUS "  KINETO_LIBRARY_TYPE = ${KINETO_LIBRARY_TYPE}")
+  message(STATUS "  CUDA_SOURCE_DIR = ${CUDA_SOURCE_DIR}")
+
+  if(EXISTS ${CUDA_SOURCE_DIR}/extras/CUPTI/lib64/libcupti_static.a)
```

Comment:
Is it possible to use CUDA_cupti_LIBRARY found by FindCUDA here? Also, see #51026, USE_CUPTI_SO should be given preference.